qcp/gates.py

```diff
@@ -277,9 +275,63 @@ def control_phase(size: int, controls: List[int], target: int,
 def phase_shift(phi: complex) -> Matrix:
     """
     Creates a 2 x 2 phase shift matrix
-
     :param complex phi: angle the qubit is phase shifted by
     returns:
         Matrix: Matrix representing the phase shift gate.
     """
     return DefaultMatrix([[1, 0], [0, cmath.exp(1j * phi)]])
+
+def swap(size: int, target: List[int]) -> Matrix: 
```

Comment:
For this `swap()` gate and the `control_U()` gate, could you just comment a link (on github) to a website/resource with some info on them, and I'll try writing some tests for them tomorrow based off of that.

Reply:
Swap gate: https://qiskit.org/textbook/ch-algorithms/quantum-fourier-transform.html#5.-The-Circuit-that-Implements-the-QFT 
Control-U: https://qiskit.org/textbook/ch-algorithms/quantum-phase-estimation.html#1.2-Mathematical-Foundation-